src/waitress/task.py

```diff
@@ -563,6 +563,10 @@ def get_environment(self):
             if mykey not in environ:
                 environ[mykey] = value
 
+        check = getattr(self.channel, 'check_client_disconnected', None)
+        if callable(check):
+            environ['check_client_disconnected'] = check
```

Comment:
when would this not be a callable?

Reply:
It was only necessary in one of the tests that used a `DummyChannel`. But I guess it is better to extend that class accordingly.

Reply:
Ah, for the tests - yeah I'd prefer to fix the tests.

The environ var is going to need to be namespaced to waitress for now. I'd call it `environ['waitress.is_connected']` or something?